### PR TITLE
[Core][DO NOT MERGE] Dummy PR to check if #58 can be compiled by Travis using forward declaration

### PR DESCRIPTION
**Pull request by loumalouomega**
This is exactly the same as #58, but using forward declaration instead of direct compilation. 

This reduces the compilation time in my computer between 30-40% (kratos_parameters is included everywhere), besides this is supposed to allow Travis to compile the library. 

This PR is basically to test that, I compiled it in my Windows machine and it compiles too, so I assume that Appveyor will not be a problem

### CODE REVIEW

**Comment by loumalouomega** [comment]
@roigcarlo same problem, could be related with the Cotire?

**Comment by philbucher** [comment]
chapeau for your persistence with this! :+1:

**Comment by loumalouomega** [comment]
> chapeau for your persistence with this! +1

imagen

**Comment by loumalouomega** [comment]
Looks like same problem

**Comment by philbucher** [comment]
It compiles!

But segfaults in the tests ...

**Comment by loumalouomega** [comment]
Looks like DEM, I never compile that application maybe is my fault, I will check

**Comment by loumalouomega** [comment]
I compiled the DEM and run everything, both release and debug, @roigcarlo any idea why this happens?

File: kratos/sources/kratos_parameters.cpp [hunk 1]
```diff
@@ -228,6 +228,14 @@ Parameters::Parameters(Parameters const& rOther)
 /***********************************************************************************/
 /***********************************************************************************/
 
+Parameters::Parameters(Parameters&& rOther) : Parameters()
+{
+    swap(rOther);
```
Comment (philbucher) [inline kratos/sources/kratos_parameters.cpp:233]: Std::

File: kratos/sources/kratos_parameters.cpp [hunk 2]
```diff
@@ -0,0 +1,1209 @@
...
+{
+    return this->GetArrayItem(Index);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Parameters& Parameters::operator=(Parameters&& rOther)
+{
+    swap(rOther);
```
Comment (philbucher) [inline kratos/sources/kratos_parameters.cpp:273]: std::

Comment (loumalouomega) [reply kratos/sources/kratos_parameters.cpp:273]: I am using the custom swap of the class

File: kratos/sources/kratos_parameters.cpp (hunk 1 above)
Comment (loumalouomega) [reply kratos/sources/kratos_parameters.cpp:233]: Idem

**Comment by loumalouomega** [comment]
Hurray it worked!!!

**Pull request closed without merging**